writer.go

```diff
@@ -443,7 +443,7 @@ func (p *MediaPlaylist) Encode() *bytes.Buffer {
 
 	head := p.head
 	count := p.count
-	for i := uint(0); i < p.winsize && count > 0; count-- {
+	for i := uint(0); (i < p.winsize || p.winsize == 0) && count > 0; count-- {
```

Comment:
If the behaviour here is to support VoD where winsize == 0, perhaps it might be better to write as:

```
	for i := uint(0); p.Closed || i < p.winsize && count > 0; count-- {
```

Then a live playlist can still support `winsize` of 0, but perhaps if the content is VoD, `winsize` should've been `count`?

I don't also fully understand (haven't looked yet), why this would get into this state, seems like if it is VoD - `winsize` should just match `count`.